briefcase/app.py

```diff
@@ -189,8 +189,11 @@ def generate_app_template(self, extra_context=None):
             template_path = os.path.expanduser('~/.cookiecutters/Python-%s-template' % self.platform)
             if os.path.exists(template_path):
                 self.template = template_path
-                self.git_checkout(template_path)
-                self.git_pull(template_path)
+                self._git_fetch(template_path)
+                self._git_checkout(template_path)
+                if not self._has_cookiecutter_json(template_path):
+                    raise BriefcaseError("Directory %r isn't a valid template (no cookiecutter.json found), try deleting'." % template_path)
```

Comment:
Let's remove the advice to try deleting the directory, I believe it will be better because I think there are many quite possible error cases when deleting won't help.